packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -62,26 +62,51 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
      * @param event - Event being logged
      * @param error - Error to extract info from
      * @param fetchStack - Whether to fetch the current callstack if error.stack is undefined
+     * @param supportsTags - Whether the event can have tagged properties (usually tied to the logger)
      */
-    public static prepareErrorObject(event: ITelemetryBaseEvent, error: any, fetchStack: boolean) {
+    public static prepareErrorObject(
+        event: ITelemetryBaseEvent,
+        error: any,
+        fetchStack: boolean,
+        supportsTags?: boolean,
+    ) {
         if (isILoggingError(error)) {
             // First, copy over stack and error message directly
             // Warning: if these were overwritten with PII-tagged props, they will be logged as-is
             const errorAsObject = error as Partial<Error>;
             event.stack = errorAsObject.stack;
             event.error = errorAsObject.message;
+            // console.log("yes, isloggingerror");
 
             // Then add any other telemetry properties from the LoggingError
-            const taggableProps = error.getTelemetryProperties();
-            for (const key of Object.keys(taggableProps)) {
+            const errorProps = error.getTelemetryProperties();
+            // console.log(errorProps);
+            for (const key of Object.keys(errorProps)) {
+                const prop = errorProps[key];
+                // console.log(`supportsTags?: ${supportsTags}`);
+                // console.log(`current prop: ${prop}`);
                 if (event[key] !== undefined) {
                     // Don't overwrite existing properties on the event
                     continue;
                 }
-                const taggableProp = taggableProps[key];
-                const { value, tag } = (typeof taggableProp === "object")
-                    ? taggableProp
-                    : { value: taggableProp, tag: undefined };
+
+                // New props not already on the event may either be tagged or untagged. Untagged props can just be
+                // directly copied over without fuss; tagged props can only be copied if policy 'supportsTags' is true.
+                // Currently, even in the case `supportsTags` is unset, we must still take care to strip away sensitive
+                // information that is marked by known tags.
+                //
+                // So, if `supportsTags` is set, just add the prop:
+                if (supportsTags) {
+                    // console.log("yes");
+                    event[key] = prop;
+                    continue;
+                }
+
+                // Otherwise, we handle the below cases in particular:
+                // Eventually supportsTags will always be set/true and this code will be removed:
+                const { value, tag } = (typeof prop === "object")
```

Comment:
Can we do more here RE object check?
For example, Si just added string array property to error object in createOdspNetworkError. While we should definitely follow up make it more type correct, we should also expect that due to bugs, laziness or other factors object payloads that are not tagged properties will end up here.

Reply:
@vladsud - I will have to look at Si's PR, `ITelemetryProperties` doesn't allow arrays so I'm suspicious.  And as for this, it's "safe" due to implementation of `LoggingError.getTelemetryProperties` which more strictly inspects any objects and only includes them if they adhere to the tagged props shape.  But that coupling is pretty loose, so good feedback.